src/gwt/src/org/rstudio/studio/client/common/filetypes/TextFileType.java

```diff
@@ -437,11 +437,13 @@ public String getScriptInterpreter()
       results.add(commands.popoutDoc());
       if (!SourceWindowManager.isMainSourceWindow())
          results.add(commands.returnDocToMain());
-      
+
       if (isR())
       {
          results.add(commands.sourceAsLauncherJob());
          results.add(commands.sourceAsJob());
+         results.add(commands.runSelectionAsJob());
+         results.add(commands.runSelectionAsLauncherJob());
```

Comment:
These commands also need to be enabled for files that can have embedded R code chunks (like R Markdown documents) -- should be a simple addition above.